test/framework/clustervalidator.go

```diff
@@ -109,79 +145,84 @@ func validateEKSAObjects(ctx context.Context, validateOpts clusterOpts) error {
+		if err := mgmtClusterClient.Get(ctx, key, u); err != nil {
 			return fmt.Errorf("cluster object does not exist %s", err)
 		}
 	}
-
+	logger.V(4).Info("EKSA objects exists validated", "cluster", clus.Name)
 	return nil
 }
 
-func validateClusterReady(ctx context.Context, validateOpts clusterOpts) error {
-	clusterClient := validateOpts.ClusterClient
-	if validateOpts.ClusterSpec.Cluster.IsManaged() {
-		clusterClient = validateOpts.ManagementClusterClient
-	}
-
-	capiCluster, err := controller.GetCAPICluster(ctx, clusterClient, validateOpts.ClusterSpec.Cluster)
+func validateClusterReady(ctx context.Context, vc ClusterValidatorConfig) error {
+	clus := vc.ClusterSpec.Cluster
+	mgmtClusterClient := vc.GetManagementClusterClient()
+	capiCluster, err := controller.GetCAPICluster(ctx, mgmtClusterClient, clus)
 	if err != nil {
 		return fmt.Errorf("failed to retrieve cluster %s", err)
 	}
-
 	if capiCluster == nil {
-		return fmt.Errorf("cluster %s does not exist", validateOpts.ClusterSpec.Cluster.Name)
+		return fmt.Errorf("cluster %s does not exist", clus.Name)
 	}
-
-	if err := checkClusterReady(capiCluster); err != nil {
-		return err
+	for _, condition := range capiCluster.Status.Conditions {
+		if condition.Type == "Ready" && condition.Status != "True" {
+			return fmt.Errorf("node %s not ready yet. %s", capiCluster.GetName(), condition.Reason)
+		}
 	}
+	logger.V(4).Info("CAPI cluster ready validated", "cluster", clus.Name)
 	return nil
 }
 
-func validateControlPlaneNodes(ctx context.Context, validateOpts clusterOpts) error {
+func validateControlPlaneNodes(ctx context.Context, vc ClusterValidatorConfig) error {
+	clus := vc.ClusterSpec.Cluster
 	cpNodes := &corev1.NodeList{}
-	if err := validateOpts.ClusterClient.List(ctx, cpNodes, client.MatchingLabels{"node-role.kubernetes.io/control-plane": ""}); err != nil {
+	if err := vc.ClusterClient.List(ctx, cpNodes, client.MatchingLabels{"node-role.kubernetes.io/control-plane": ""}); err != nil {
 		return fmt.Errorf("failed to list controlplane nodes %s", err)
 	}
 
-	cpConfig := validateOpts.ClusterSpec.Cluster.Spec.ControlPlaneConfiguration
+	cpConfig := clus.Spec.ControlPlaneConfiguration
 	if len(cpNodes.Items) != cpConfig.Count {
 		return fmt.Errorf("control plane node count does not match expected: %v of %v", len(cpNodes.Items), cpConfig.Count)
 	}
 
 	for _, node := range cpNodes.Items {
-		if err := validateNodeReady(node, validateOpts.ClusterSpec.Cluster.Spec.KubernetesVersion); err != nil {
+		if err := validateNodeReady(node, clus.Spec.KubernetesVersion); err != nil {
 			return fmt.Errorf("failed to validate controlplane %s", err)
 		}
 	}
+	logger.V(4).Info("Control plane nodes validated", "cluster", clus.Name)
 	return nil
 }
 
-// TODO: Update this to also validate BottleRocket nodes
-func validateWorkerNodes(ctx context.Context, validateOpts clusterOpts) error {
-	var workerNodes []corev1.Node
+func validateWorkerNodes(ctx context.Context, vc ClusterValidatorConfig) error {
```

Comment:
Are we removing the worker node count validation?

Reply:
Ah, thanks for catching that. Wasn't my intention, I'll add that back in.